vsc_pdf_template_transformer/lib/src/vsc_pdf_template_transformer_base.dart

```diff
@@ -27,12 +28,23 @@ class Transformer {
   }
 
   //TODO: Extend with more types. Mirrors package not available in Flutter project so we can't create an instance of a class by name via reflection
-  static WidgetBuilder? getWidgetBuilder(
-      Map<String, dynamic> valueMap, Map<String, dynamic> data) {
+  static WidgetBuilder? getWidgetBuilder(Map<String, dynamic> valueMap) {
     WidgetBuilder? result;
-
-    const widgetClassFromJson = {
-      'TplText': TplText.fromJson,
+    final className = valueMap['className'];
+    Function? f;
+    switch (className) {
```

Comment:
There was previously a map here that I asked the previous dev to use instead of a switch. It's one line per widget instead of 3, and it can be a `const` defined outside of this method. For example:

```dart
const widgetClassFromJson = {
    'TplText': TplText.fromJson,
    'TplSizedBox': TplSizedBox.fromJson,
    'TplContainer': TplContainer.fromJson,
};
```

We're going to have a lot of widgets, so using a switch will get to be a large amount of code.